### PR TITLE
make brightness control via hotkeys exponential

### LINKED ISSUE #112: make brightness control exponential
As brightness is perceived exponentially and not linearly, it would be more intuitive for brightness control via hotkeys to follow an exponential curve as well. `brightnessctl` supports this already by setting the `-e` flag.

### PR DESCRIPTION
**Pull request by nova-r**
fixes #112 
As brightness is perceived exponentially, it is more intuitive to make the brightness change exponentially as well.

### CODE REVIEW

File: .config/sway/config.d/50-openSUSE.conf [hunk 1]
```diff
@@ -37,8 +37,8 @@ bindsym --to-code {
 # Media keys
 bindsym XF86AudioMicMute exec pactl set-source-mute @DEFAULT_SOURCE@ toggle
 
-bindsym XF86MonBrightnessDown exec brightnessctl -q set 5%- && ( echo $((`brightnessctl get` * 100 / `brightnessctl m`)) > $SWAYSOCK.wob )
-bindsym XF86MonBrightnessUp exec brightnessctl -q set +5% && ( echo $((`brightnessctl get` * 100 / `brightnessctl m`)) > $SWAYSOCK.wob )
+bindsym XF86MonBrightnessDown exec brightnessctl -q -e set 5%- && ( perl -e "use Math::Complex; printf \"%.0f\n\", 100 * ((`brightnessctl get` / `brightnessctl m`)**0.25)" > $SWAYSOCK.wob )
```
Comment (denisok) [inline .config/sway/config.d/50-openSUSE.conf:40]:
```suggestion
bindsym XF86MonBrightnessDown exec brightnessctl -q -e set 5%- && ( perl -e 'use Math::Complex; printf \"%.0f\n\", 100 * ((`brightnessctl get` / `brightnessctl m`)**0.25)' > $SWAYSOCK.wob )
```

File: .config/sway/config.d/50-openSUSE.conf [hunk 2]
```diff
@@ -37,8 +37,8 @@ bindsym --to-code {
...
+bindsym XF86MonBrightnessUp exec brightnessctl -q -e set +5% && ( perl -e "use Math::Complex; printf \"%.0f\n\", 100 * ((`brightnessctl get` / `brightnessctl m`)**0.25)" > $SWAYSOCK.wob )
```
Comment (denisok) [inline .config/sway/config.d/50-openSUSE.conf:41]:
```suggestion
bindsym XF86MonBrightnessUp exec brightnessctl -q -e set +5% && ( perl -e 'use Math::Complex; printf \"%.0f\n\", 100 * ((`brightnessctl get` / `brightnessctl m`)**0.25)' > $SWAYSOCK.wob )
```

**Review by denisok — CHANGES_REQUESTED** [review]
Otherwise doesn't work for me:
```
May 22 11:52:37 stdlen sway[4702]: Execution of -e aborted due to compilation errors.
May 22 11:52:37 stdlen sway[4702]: syntax error at -e line 1, near "%.0"
May 22 11:52:37 stdlen sway[4702]:         (Missing operator before f?)
May 22 11:52:37 stdlen sway[4702]: Bareword found where operator expected at -e line 1, near "0f"
May 22 11:52:37 stdlen sway[4702]:         (Missing operator before 0?)
May 22 11:52:37 stdlen sway[4702]: Number found where operator expected at -e line 1, near "%.0"
```

**Comment by denisok** [comment]
@nova-r could you please check suggestions?

**Comment by denisok** [comment]
@FilippoBonazziSUSE could you please check this one. When I checked it last time, it didn't work for me.

**Review by FilippoBonazziSUSE — CHANGES_REQUESTED** [review]
The commands work for me (with the necessary `'` to `"` fix) in the sense that they succeed in changing brightness.

However, there are 3 things that I would like to see changed:

1. The use of the Perl interpreter is not ideal, and I'd like to avoid it if at all possible. Any extra dependency must be carefully considered
2. The brightness up and down commands are not symmetrical, i.e. increments on the way up are different from decrements on the way down. This makes brightness values not reproducible and arbitrary looking. For example, starting from 20%, going up to 100% takes N increments, but going back down N decrements arrives at 19%. This needs to be fixed before this can be merged.
3. There is no progress on the discussion we had in #112 , and I'd like to see those points addressed too

**Comment by FilippoBonazziSUSE** [comment]
@denisok This feature will take a bit more time, let's not wait for this for 0.16.0

File: .config/sway/config.d/50-openSUSE.conf (hunk 1 above)
Comment (mcepl) [inline .config/sway/config.d/50-openSUSE.conf:40]: Pulling in yet another programming language (Perl) is not a good idea. Look at https://github.com/openSUSE/openSUSEway/issues/126 as well.

**Review by mcepl — CHANGES_REQUESTED** [review]

**Pull request closed without merging**